chsdi/templates/luftbilder/viewer.mako

```diff
@@ -260,7 +288,118 @@
         view.on('propertychange', debounce(updateUrl));
         updateUrl();
 
+        var contrastSlider = document.getElementById("contrast");
+        var contrastOut = document.getElementById("contrastOut");
+        var resetbutton = document.getElementById("reset");
+        var minusbutton = document.getElementById("minus-button");
+        var plusbutton = document.getElementById("plus-button");
+
+        setSliderListeners();
+
+        var equalizedValues = undefined;
+        var mean = undefined;
+
+        function getPixelList(data){
+          var pixel_list = [];
+          for (var i=0; i<(data.length);i+=4){
+            if (data[i+3]!=0){
+              var pixel = [data[i], data[i+1], data[i+2], data[i+3]];
+              pixel_list.push(pixel);
+            }
+          }
+          return pixel_list;
+        }
+
+        function allZero(data){
+          for (var key in data){
+            if (data[key] != 0) return false
+          }
+          return true
+        }
+
+        // computes normalized histogram
+        function getHistogramProperties(pixels){
+          var histogram = {};
+          var equalizedHistogram = {};
+          var number_pixels = pixels.length;
+          var sum = 0;
+          var mean_luminance = 0;
+
+          //initialize histogram 
+          for (var i = 0; i < 101; i++){
+            histogram[i] = 0;
+          }
+
+          for (var p in pixels){
+            // convert to hcl
+            var pix = pixels[p];
+            var hcl = rgb2hcl(pix);
+            // get luminance values
+            var l = parseInt(hcl[2]);
+            // augment count
+            histogram[l] = histogram[l] + 1;
+          }
+
+          //normalization of histogram + compute equalized Values and Mean
+          for (var i in histogram){
+            histogram[i] = histogram[i] / number_pixels;
+            sum += histogram[i];
+            equalizedHistogram[i] = sum;
+            mean_luminance += histogram[i]*i;
+          }
+          return [equalizedHistogram, mean_luminance];
+        }
+
+        lubisMap.on('postrender', function(event){
+          var canvas = document.getElementsByTagName('canvas')[0];
+          var context = canvas.getContext('2d');
+          var imageData = context.getImageData(0,0, canvas.width, canvas.height).data;
+          // to prevent that recomputed every time map rendered
+          if (!allZero(imageData) && !mean) {
+            var pixel_list = getPixelList(imageData);
+            var histogramProperties = getHistogramProperties(pixel_list);
+            equalizedValues = histogramProperties[0];
+            mean = histogramProperties[1];
+          }
+        });
+
+        raster.on('beforeoperations', function(event) {
+          var data = event.data;
+          data["contrast"] = Number(contrastSlider.value);
```

Comment:
The value is already a number no?

Reply:
I initially thought the same, but it's a String.

Reply:
ok use `parseInt` as you do in the other case